### ISSUE TITLE
How does K8s decide to use ovn as a cni?

### ISSUE THREAD

**Issue by yvzsrt** [openvswitch/ovn-kubernetes#426]
We're trying to setup ovn with kubernetes. There are two points that we couldn't find from documentation or couldn't understand we want to clarify.

First, when setting up kubernetes, a cni module (such as calico, flannel) should be installed after init command. Does installing and setting up OVN come after this process? Do we install ovn after we have a stable k8s setup with a network cni like flannel?

Second, K8s reads the configuration files under /etc/cni/net.d to decide which cni module to run. Does ovn have a conf file under that directory? How does K8s decide to use ovn as network module?

Sorry if i missed something in documentation.

**Comment by alinbalutoiu** [comment]
Hi, the CNI config file gets automatically generated by ovnkube, you just need to specify the path where you want to have it created. (https://github.com/openvswitch/ovn-kubernetes/blob/master/docs/ovnkube.1#L84-L85)

I am not sure to what "init command" you are referring, but ovn-kubernetes is like any other CNI plugin. K8s will read the config generated by ovnkube under /etc/cni/net.d.

You can follow the documentation from here: https://github.com/openvswitch/ovn-kubernetes#master-node-initialization

In short, on master you have to:
- Start the central components on a k8s master node.
- Start ovn-northd and ovn-controller
- Run ovnkube command

On minion:
- Start kubelet
- Start ovn-controller
- Run ovnkube command

You can also refer to the ansible playbooks if you want to see how the configuration is being done: https://github.com/openvswitch/ovn-kubernetes/tree/master/contrib

**Comment by yvzsrt** [comment]
Thanks for your help. We're trying vagrant and ansible solutions.
